### PR TITLE
Email oauth2

### PR DESCRIPTION
**Pull request by jasiek-net**
#29

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -11,9 +11,10 @@ DB_PORT=5432
...
-MAIL_SERVICE=gmail
-MAIL_USER=viking.garage.app@gmail.com
-MAIL_PASS=secret
+GMAIL_USER=viking.garage.app@gmail.com
+GMAIL_CLIENT_ID=secret
+GMAIL_CLIENT_SECRET=secret
+GMAIL_REFRESH_TOKEN=secret
 
 # JSON Web Token config
 JWT_SECRET=secret-jwt-key
```
Comment (michalmikolajczyk) [inline .env.example:20]: is there a way we can rewrite the history for this file?

File: src/mailer/index.ts [hunk 2]
```diff
@@ -3,15 +3,26 @@ import {
...
 } from './templates';
-const ride = process.env.VG_CONTACT_EMAIL || 'viking.garage.app@gmail.com';
-const config = {
-  service: process.env.MAIL_SERVICE,
+
+if (!process.env.GMAIL_USER ||
+    !process.env.GMAIL_CLIENT_ID ||
+    !process.env.GMAIL_CLIENT_SECRET ||
+    !process.env.GMAIL_REFRESH_TOKEN) {
+  console.log('You should set Gmail OAuth2 envs');
```
Comment (michalmikolajczyk) [inline src/mailer/index.ts:11]: let's make this a debug log

File: src/mailer/index.ts [hunk 3]
```diff
@@ -3,15 +3,26 @@ import {
...
+}
+
+const transporter = nodemailer.createTransport({
```
Comment (michalmikolajczyk) [inline src/mailer/index.ts:14]: can we capture the error within the assignment, using lodash perhaps?

File: src/mailer/index.ts [hunk 4]
```diff
@@ -25,7 +36,7 @@ function sendEmail(body: any): Promise<any> {
 
 function signinEmail(name: string, email: string, token: string, code: string): Promise<any> {
   const body = {
-    from: config.auth.user,
+    from: process.env.GMAIL_USER,
```
Comment (michalmikolajczyk) [inline src/mailer/index.ts:39]: perhaps we could make the assignment at the top, so we don't have to refer to the `process` object every time and instead have a regular `gmailUser` const

Comment (michalmikolajczyk) [reply src/mailer/index.ts:39]: or `config.gmailUser`

**Review by michalmikolajczyk — APPROVED** [review]

File: .env.example (hunk 1 above)
Comment (jasiek-net) [reply .env.example:20]: could you explain?

File: src/mailer/index.ts (hunk 3 above)
Comment (jasiek-net) [reply src/mailer/index.ts:14]: this one doesn't return any error (with wrong configuration error will be on `transporter.sendMail()`)

**Pull request merged**